sr_port/op_fnzpiecediff.c

```diff
@@ -0,0 +1,620 @@
+        {
+          if (*arg2_p == *delim_p)
+          {
+            if (ans_index==0) 
+            {
+              answer[ans_index] = curr_section + '0';
+              ans_index++;
+            } else 
+            {
+              answer[ans_index] = ',';
+              ans_index++;
+              
+              section_len = 0;
+              temp_section = curr_section;
+              while (temp_section) 
+              {
+                section_len++;
+                temp_section /= 10;
+              }
+              temp_section_len = section_len;
+              temp_section = curr_section;
+              while (temp_section_len)
+              {
+                answer[ans_index+temp_section_len-1] = (temp_section % 10) + '0';
+                temp_section /= 10;
+                temp_section_len--;
+              }
+              ans_index += section_len;
+            }
+            curr_section++;
+          }
+          arg2_i++;
+          arg2_p++;
+        }
+        if (ans_index==0) 
+        {
+          answer[ans_index] = curr_section + '0';
+          ans_index++;
+        } else 
+        {
+          answer[ans_index] = ',';
+          ans_index++;
+          
+          section_len = 0;
+          temp_section = curr_section;
+          while (temp_section) 
+          {
+            section_len++;
+            temp_section /= 10;
+          }
+          temp_section_len = section_len;
+          temp_section = curr_section;
+          while (temp_section_len)
+          {
+            answer[ans_index+temp_section_len-1] = (temp_section % 10) + '0';
+            temp_section /= 10;
+            temp_section_len--;
+          }
+          ans_index += section_len;
+        }
+        break;
+      }
+      if (arg2_i >= arg2_len)
+      {
+        if (*arg1_p == *delim_p && !found_end)
+        {
+          arg1_i++;
+          arg1_p++;
+          curr_section++;
+          found_end = 1;
+        }
+        while (arg1_i < arg1_len)
+        {
+          if (*arg1_p == *delim_p)
+          {
+            if (ans_index==0) 
+            {
+              answer[ans_index] = curr_section + '0';
+              ans_index++;
+            } else 
+            {
+              answer[ans_index] = ',';
+              ans_index++;
+              
+              section_len = 0;
+              temp_section = curr_section;
+              while (temp_section) 
+              {
+                section_len++;
+                temp_section /= 10;
+              }
+              temp_section_len = section_len;
+              temp_section = curr_section;
+              while (temp_section_len)
+              {
+                answer[ans_index+temp_section_len-1] = (temp_section % 10) + '0';
+                temp_section /= 10;
+                temp_section_len--;
+              }
+              ans_index += section_len;
+            }
+            curr_section++;
+          }
+          arg1_i++;
+          arg1_p++;
```

Comment:
This section (141-164) seems to be repeated a bit; consider moving it to a macro